dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/persistence/datastore/DatastoreFileWriter.kt

```diff
@@ -85,6 +86,18 @@ internal class DatastoreFileWriter(
         }
     }
 
+    @WorkerThread
+    internal fun clearAllData() {
+        val dataStoreDirectory = dataStoreFileHelper.getDataStoreDirectory(
+            featureName = featureName,
+            storageDir = storageDir
+        )
+
+        if (dataStoreDirectory.existsSafe(internalLogger)) {
+            dataStoreDirectory.deleteDirectoryContentsSafe(internalLogger)
```

Comment:
there is no need to introduce `deleteDirectoryContentsSafe` method, you can simply do `dataStoreDirectory.deleteSafe` here (one call to delete the whole directory instead of many calls to delete each file).

```suggestion
            dataStoreDirectory.deleteSafe(internalLogger)
```

Is there any reason to delete individual files instead of the whole directory?